matrix/matrix_binary_search.py

```diff
@@ -0,0 +1,54 @@
+def binary_search(array: list, lower_bound: int, upper_bound: int, value: int) -> int:

```

Comment:
```suggestion
def binary_search(array: list, value: int) -> int:
```
I don't think `lower_bound` and `upper_bound` parameters do much. If the person wants to change the upper and lower bound they should pass the list indexed to the upper and lower bound, such as 

`>>> binary_search([1,2,3,4,5][1:3])`